### PR TITLE
config, executor, sessionctx/variable: Remove old config items

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/34867 close https://github.com/pingcap/tidb/issues/34945 close https://github.com/pingcap/tidb/issues/34879

Problem Summary:

As we have removed certain config settings, but kept them in the config structure for import purposes it gets confusing when inspecting meta-data commands. This introduces a feature to patch-out the removed items from `select @@tidb_config` and infoschema tables.

The "deprecatedConfig" was misnamed, since these options are not really deprecated but actually removed. Or rather it was ambiguous, because some were actually deprecated and still took effect.

### What is changed and how it works?

Meta data commands like `show config` no longer show removed configuration settings.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Meta data commands like `show config` no longer show removed configuration settings.
```

### CODE REVIEW

**Review by zimulala — APPROVED** [review]
LGTM

**Comment by zimulala** [comment]
PTAL @bb7133

File: config/config.go [hunk 1]
```diff
@@ -942,8 +946,6 @@ var deprecatedConfig = map[string]struct{}{
 	"max-txn-time-use":                   {},
 	"experimental.allow-auto-random":     {},
 	"enable-redact-log":                  {}, // use variable tidb_redact_log instead
-	"tikv-client.copr-cache.enable":      {},
-	"alter-primary-key":                  {}, // use NONCLUSTERED keyword instead
```
Comment (mjonss) [inline config/config.go:946]: I probably miss some context, but why are these removed?

**Review by mjonss — APPROVED** [review]
LGTM

Just a small question :)

File: config/config.go (hunk 1 above)
Comment (morgo) [reply config/config.go:946]: > "The "deprecatedConfig" was misnamed, since these options are not really deprecated but actually removed. Or rather it was ambiguous, because some were actually deprecated and still took effect."

They are deprecated, they both still have usage. I initially split this into a map for deprecated and a map for removed, and even wrote tests to make sure they were not in both lists :-) But actually, there is no code that cares about deprecated to do anything different. So I just removed them from the list and renamed it.

**Comment by morgo** [comment]
@bb7133 I need your review as code owner

**Comment by morgo** [comment]
@zimulala @mjonss I've modified this to include a fix for https://github.com/pingcap/tidb/issues/34879 as well. PTAL again :-)

**Review by bb7133 — APPROVED** [review]
LGTM
